src/test/java/teammates/test/driver/Retryable.java

```diff
@@ -0,0 +1,42 @@
+package teammates.test.driver;
+
+/**
+ * Represents a task that can be retried.
+ * @param <T> Result type.
+ * @param <E> Throwable type.
+ */
+public abstract class Retryable<T, E extends Throwable> {
+
+    protected String name;
+
+    public Retryable(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Runs the task once and returns the result.
+     */
+    // CHECKSTYLE.OFF:MethodName internal variable underscore naming convention applied to abstract methods
+    protected abstract T run_() throws E;
+
+    /**
+     * Checks whether the task succeeded.
+     */
+    protected abstract boolean isSuccessful_() throws E;
+    // CHECKSTYLE.OFF:MethodName
```

Comment:
I'd rather avoid it if possible. It's an old convention that doesn't seem to be used these days among Java users.  :-p